lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -39,9 +41,16 @@ def collect(self, module=None, collected_facts=None):
             if rc != 0:
                 return systemd_facts
 
-            systemd_facts["systemd"] = {
-                "features": str(stdout.split("\n")[1]),
-                "version": int(stdout.split(" ")[1]),
-            }
+            systemd_version = stdout.split(" ")[1]
+
+            systemd_facts["systemd"] = {}
+            systemd_facts["systemd"]["features"] = stdout.split("\n")[1]
+            systemd_facts["systemd"]["full_version"] = str(
+                stdout.split(" ")[2].split(")")[0][1:],
```

Comment:
If I'm reading it right, it's already a string:
```suggestion
            systemd_facts["systemd"]["full_version"] = (
                stdout.
                split(" ")[2].
                split(")")[0][1:]
```

Reply:
@webknjaz well yes, but actually no. If you build eg https://github.com/systemd/systemd/releases/tag/v257 it will be an integer, but in the absolute majority of cases if will be a string.

```sh
$ systemctl --version && ansible -i'localhost,' -c local -m setup -a 'filter=systemd' localhost 2>/dev/null
systemd 257 (257)                                                                                                                                                                                                +PAM +AUDIT +SELINUX +APPARMOR +IMA +IPE +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 -PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -BPF_FRAMEWORK -BTF -XKBCOMMON +UTMP +SYSVINIT -LIBARCHIVE
localhost | SUCCESS => {
    "ansible_facts": {
        "ansible_systemd": {
            "features": "+PAM +AUDIT +SELINUX +APPARMOR +IMA +IPE +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 -PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -BPF_FRAMEWORK -BTF -XKBCOMMON +UTMP +SYSVINIT -LIBARCHIVE",
            "full_version": "257",
            "version": 257
        },
        "discovered_interpreter_python": "/usr/bin/python3.12"
    },                                                                                                                                                                                                               "changed": false
}
```

```sh
$ systemctl --version && ansible -i'localhost,' -c local -m setup -a 'filter=systemd' localhost 2>/dev/null
systemd 256 (256.5-2ubuntu3.1)
+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT +LIBARCHIVE
localhost | SUCCESS => {
    "ansible_facts": {
        "ansible_systemd": {
            "features": "+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT +LIBARCHIVE",
            "full_version": "256.5-2ubuntu3.1",
            "version": 256
        },
        "discovered_interpreter_python": "/.../Code/venvs/ansible/bin/python3.12"
    },
    "changed": false
}
```